api/v1alpha1/provisioning_validation.go

```diff
@@ -0,0 +1,206 @@
+package v1alpha1
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var (
+	log = ctrl.Log.WithName("provisioning_validation")
+)
+
+/*
+   Validate the contents of the provisioning resource
+
+   Managed:
+   "ProvisioningInterface"
+   "ProvisioningIP"
+   "ProvisioningNetworkCIDR"
+   "ProvisioningDHCPRange"
+   "ProvisioningOSDownloadURL"
+
+   Unmanaged:
+   "ProvisioningInterface"
+   "ProvisioningIP"
+   "ProvisioningNetworkCIDR"
+   "ProvisioningOSDownloadURL"
+
+   Disabled:
+   "ProvisioningOSDownloadURL"
+*/
+func (prov *Provisioning) Validate() error {
+	provisioningNetworkMode := prov.getProvisioningNetworkMode()
+	log.V(1).Info("provisioning network", "mode", provisioningNetworkMode)
+
+	var errs []error
+
+	// They all use provisioningOSDownloadURL
+	if err := validateProvisioningOSDownloadURL(prov.Spec.ProvisioningOSDownloadURL); err != nil {
+		errs = append(errs, err...)
+	}
+
+	// Disabled does not need to check network settings.
+	if provisioningNetworkMode == ProvisioningNetworkDisabled {
+		return errors.NewAggregate(errs)
+	}
```

Comment:
mmm, is this correct? I thought that now they are just optional. so validate if provided?

Reply:
Yep.  Fixing now.